### ISSUE TITLE
`from_npz()` method for loading

### ISSUE THREAD

**Issue by vigji** [pynapple-org/pynapple#169]
As far as I could find in the docs, there is a convenient method for saving a Tsd as a npc file, but putting it back together would always require to do:

```python
file = np.load("my_path/my_tsd.npz")
time_support = nap.IntervalSet(file['start'], file['end'])
nap.Tsd(t=file['t'], d=file['d'], time_support=time_support)
```
which is not that much work, but it would be probably better to have a class method like (similarly to pandas from_csv()):

```python
class Tsd:
    ...
    @classmethod
    def from_npz(cls, file_path):
        file = np.load(file_path)
        time_support = nap.IntervalSet(file['start'], file['end'])
        return cls(t=file['t'], d=file['d'], time_support=time_support)

```

So that the interface would be super simple:
```python
loaded_obj = Tsd.from_npz("my_path/my_tsd.npz")
```

What would you think about that?

**Comment by gviejo** [comment]
The idea was to use [nap.load_file(filename)](https://pynapple-org.github.io/pynapple/reference/io/misc/#pynapple.io.misc.load_file) but yes I agree this could be an extra convenient way for it.

**Comment by vigji** [comment]
the advantage would be to keep all the io-related naming conventions in a single place! (and a oo interface). Can do a PR with tests

**Comment by tsloan1377** [comment]
I'm similarly interested in being able to load LFP, spike, and behavior data that is already processed and saved as numpy arrays and csv files. 
From what I can see in the documentation, most of the I/O functionality is set up to load from the BIDS format as in the example notebooks/data.

Since this issue is still open an option to directly import from numpy files is of interest, or in the works. 

As of right now, what would be the path of least resistance to have a numpy arrays and pandas dataframes of data wrapped into pynapple data structures? Renaming my files and folder structure to the BIDS specification outlined [here](https://bids-standard.github.io/bids-starter-kit/folders_and_files/folders.html)? Or directly wrangling the datatypes into the same format as data = nap.load_session() provides?

Thanks in advance for your advice on the matter.

**Comment by vigji** [comment]
This has now been addressed in #313, even if the method is not public the logic is there
